Source/Diagnostics/Diagnostics.H

```diff
@@ -62,6 +62,15 @@ public:
      * \param[in] lev level on which the vector of unique_ptrs to field functors is initialized.
      */
     virtual void InitializeFieldFunctors (int lev) = 0;
+    /** Initialize functors that store pointers to the fields requested by the user.
+     *
+     * Derived classes MUST implement this function, and it must allocate m_all_field_functors
```

Comment:
```suggestion
     * Derived classes MUST implement this function, and it must allocate m_all_field_functors
```

if you like to force that this must be implemented, then declare this purely virtual:
```C++
virtual void InitializeFieldFunctorsRZ (int lev) = 0;
```

Otherwise, it's currently implemented to do nothing, unless overwritten.

Reply:
I wasn't going to define back-transformed InitializeFieldFunctorsRZ in this PR, so I wasn't sure whether to leave an empty implementation in Diagnostics.H or to force an empty implementation in child classes of Diagnostics

Reply:
Ok, then maybe change the docs to: `Derived classes can implement this function for RZ. If so, it must allocate m_all_field_functors.`